Development branch

itunuworks opened the pull request:
Kindly help review this against the project requirements. Which can be found [here](https://docs.google.com/a/andela.com/document/d/1yjxqITDFZuKgdWyHl-6hGPoYfja8Z_BpyPEkd2dhGYI/edit?usp=sharing)

ifatoki committed:
Complete project file system setup and initialization

ifatoki committed:
setup webpack config and basic html. Also implemented some Actions, dispatcher and the store.

ifatoki committed:
Implement a new Component DropdownItem and Render it with Sources

ifatoki committed:
Fix flux generation of Sources and Articles

ifatoki committed:
Fix the autopopulating dropdownbox

ifatoki committed:
Refactored code using 'article' instead of 'headline' for better clarity

ifatoki committed:
Complete view articles based on selection and search through selection

ifatoki committed:
Implement reading of news article in new tab based on click event

itunuworks committed:
Merge pull request #1 from itunuworks/projectSetupBranch

Complete project file system setup and initialization

itunuworks committed:
Merge pull request #4 from itunuworks/basicFunctionalityBranch

Basic functionality branch

ifatoki committed:
Make modifications to loading process of page

ifatoki committed:
Setup filter dropdown and functionality

ifatoki committed:
Implement population of filter dropdown based on source selected

itunuworks committed:
Merge pull request #5 from itunuworks/redesignPageLoadSetupBranch

Redesign page load setup branch

ifatoki committed:
Setup Google+ authentication using react-google-login

ifatoki committed:
Enforce reseting of filter selection to default on source change

ifatoki committed:
Begin implementation of tests

itunuworks committed:
Merge pull request #6 from itunuworks/googleSignInBranch

Google sign in branch

Commit:
Setup basic routing system and remove unnecessary console messages

Commit:
Implement eslint, fix lint errors in code

Commit:
Implement some code refactoring

itunuworks committed:
Install node6 transpiler, fix tests, modify setup of webpack to include the node 6

itunuworks committed:
Setup Babel and Jest testing configuration and run a simple sample test successfully

itunuworks committed:
Refactor some lines of code in Sources.js

itunuworks committed:
Refactor and clear out eslint errors

itunuworks committed:
Merge pull request #7 from itunuworks/testsSetupBranch

Tests setup branch

itunuworks committed:
Implement basic signIN and signOut function flow.

itunuworks committed:
Implement server.js, login and logout

itunuworks committed:
Correct run error in package.json

itunuworks committed:
Perfect routing and have it work well.

Commit:
Refactor routing

Commit:
Incorporate Materialize with UI, Implement code commenting and refactor code extensively

itunuworks committed:
Merge pull request #8 from itunuworks/firebaseSetupBranch

Firebase setup branch

Commit:
Setup travisCI

Commit:
Modify .travis.yml

itunuworks committed:
Merge pull request #9 from itunuworks/firebaseSetupBranch

Setup travisCI complete and all tests passed.

Commit:
Setup Hound and extend the constraints on eslint to Andela standards

itunuworks committed:
Merge pull request #10 from itunuworks/firebaseSetupBranch

Setup Hound and extend the constraints on eslint to Andela standards

Commit:
Fix eslint errors

Commit:
Refactored some code

itunuworks committed:
Update firebaseApi.js

itunuworks committed:
Merge pull request #11 from itunuworks/firebaseSetupBranch

Firebase setup branch

Commit:
Make additional tests and includes travis badge in readme

Commit:
Merge branch 'firebaseSetupBranch' of https://github.com/itunuworks/news-feed-application into firebaseSetupBranch

THis is necessary to place the remote and local repo at par.

itunuworks committed:
Merge pull request #12 from itunuworks/firebaseSetupBranch

Firebase setup branch

Commit:
Modify README and test

Commit:
Fix chmodifications to the ListView component and Parent

itunuworks committed:
Merge pull request #13 from itunuworks/firebaseSetupBranch

Firebase setup branch

ifatoki committed:
Remove local materialize files and use CDN, begin UI setup

itunuworks committed:
Merge pull request #14 from itunuworks/additionalFeaturesBranch

Remove local materialize files and use CDN, begin UI setup

ifatoki committed:
Delete local materialize files

Commit:
Complete coveralls setup and compile ready packed file

itunuworks committed:
Merge pull request #15 from itunuworks/additionalFeaturesBranch

Additional features branch

Commit:
Modify test setup

itunuworks committed:
Merge pull request #16 from itunuworks/additionalFeaturesBranch

Modify test setup

Commit:
Setup store tests with 100% coverage

Commit:
store test mocked and passes 100%

Commit:
Fix tests for newsAction with 71.93% coverage

itunuworks committed:
Merge pull request #17 from itunuworks/setupTestsBranch

Setup tests branch

Commit:
Configure coveralls

Commit:
REadme update

Commit:
REadme update

Commit:
Merge branch 'setupTestsBranch' of https://github.com/itunuworks/news-feed-application into setupTestsBranch

itunuworks committed:
Merge pull request #18 from itunuworks/setupTestsBranch

Setup tests branch

itunuworks committed:
Update README.md

itunuworks committed:
Update README.md

Commit:
Merge branch 'setupTestsBranch' of https://github.com/itunuworks/news-feed-application into setupTestsBranch

itunuworks committed:
Merge pull request #19 from itunuworks/setupTestsBranch

Setup tests branch

Commit:
Make modifications to coveralls badge

itunuworks committed:
Merge pull request #20 from itunuworks/setupTestsBranch

Make modifications to coveralls badge

Commit:
Make a quick improvement to the links in NewsItem

itunuworks committed:
Merge pull request #21 from itunuworks/setupTestsBranch

Make a quick improvement to the links in NewsItem

Commit:
Setup the DropDowns

Commit:
Setup Main page with SemanticUI

Commit:
Fix routing challenges

Commit:
Modify Dropdowns

Commit:
Fix binding challenges in Source component

Commit:
Wrapup basic UI

itunuworks committed:
Merge pull request #22 from itunuworks/semanticUIBranch

Semantic ui branch

itunuworks committed:
Merge pull request #23 from itunuworks/routingBranch

Fix routing challenges

Commit:
Fix local js files instead of CDN

itunuworks committed:
Merge pull request #24 from itunuworks/fixWebPack

Fix local js files instead of CDN

Commit:
Fix the break in the header rerouting protocol.

itunuworks committed:
Merge pull request #25 from itunuworks/fixBugOnNewsHeaderRedirect

Fix the break in the header rerouting protocol.

Commit:
Update the webpack file

itunuworks committed:
Merge pull request #27 from itunuworks/fixBugOnNewsHeaderRedirect

Update the webpack file

Commit:
Fix the basic README file

itunuworks committed:
Merge pull request #30 from itunuworks/readmeBranch

Fix the basic README file

freemile2ru commented on .hound.yml:
```diff
@@ -0,0 +1,30 @@
+  enabled: true
+  config_file: .eslintrc.json
+
+ruby:
```
this aint necessary here. just the first  3 lines are sufficient

freemile2ru commented on __tests__/actions/articles.json:
```diff
@@ -0,0 +1,39 @@
+{
```
what is with the folder naming convention **__tests__**  please check your style guides again

freemile2ru commented on spec/sum.js:
```diff
@@ -0,0 +1,5 @@
+function sum(a, b) {
```
use arrow function instead

freemile2ru commented on src/js/actions/newsActions.js:
```diff
@@ -0,0 +1,48 @@
+import firebaseApi from '../api/firebaseApi';
+
+// Handle the signIn event.
+export function signIn() {
```
arrow functions?

freemile2ru commented on src/js/actions/newsActions.js:
```diff
@@ -0,0 +1,48 @@
+}
+
+// Recieve authStateChanged event from the firebaseApi and resolve the new user.
+export function authStateChangedHandler(user) {
```
arrow functions?

freemile2ru commented on src/js/actions/newsActions.js:
```diff
@@ -0,0 +1,48 @@
+same once api is done fetching.
+Also dispatch the GET_ARTICLES message and fetched articles.
+*/
+export function getArticles(source, sortBy) {
```
arrow functions?

freemile2ru commented on src/js/actions/newsActions.js:
```diff
@@ -0,0 +1,48 @@
+Request for sources from the NewsApi and recieve same once api is done fetching.
+Also dispatch the GET_SOURCES message and fetched sources.
+*/
+export function getSources() {
```
arrow functions?

freemile2ru commented on src/js/api/newsAPI.js:
```diff
@@ -0,0 +1,46 @@
+      });
+  }
+}
+
```
what is going on here? jsdocs please

Commit:
Implement JSDOC across files

itunuworks committed:
Merge pull request #32 from itunuworks/jDocBranch

Implement JSDOC across files

Commit:
Fix routing bug

Commit:
Fix several eslint errors

freemile2ru commented on src/js/api/firebaseApi.js:
```diff
@@ -0,0 +1,96 @@
+  signIn() {
+    this.firebase.auth().signInWithRedirect(this.provider)
+    .then((result) => {
+      console.log(result);
```
delete console.log from all code before pushing

freemile2ru commented on src/js/api/firebaseApi.js:
```diff
@@ -0,0 +1,96 @@
+  signOut() {
+    this.firebase.auth().signOut()
+      .then((result) => {
+        console.log(result);
```
delete console.log from all code before pushing

freemile2ru commented on src/js/api/firebaseApi.js:
```diff
@@ -0,0 +1,96 @@
+    console.log(`The article with url: ${url} is added to list: ${listName}`);
+  }
+
+  deleteUserList(listName) {
```
why commented out code?

freemile2ru commented on src/js/api/newsAPI.js:
```diff
@@ -0,0 +1,73 @@
+import rp from 'request-promise';
```
what is rp? please variable names should be clear and descriptive

freemile2ru commented on src/js/components/NewsItem.jsx:
```diff
@@ -0,0 +1,61 @@
+   * @function handleClick
+   * @param {object} e
+   */
+  function handleClick(e) {
```
what is e?, please check style guide

freemile2ru commented on src/js/pages/Home.jsx:
```diff
@@ -0,0 +1,18 @@
+ * @function Home
+ * @returns {JSX Component}
+ */
+export default function Home() {
```
use arrow functions

Commit:
Make improvements to the UI

Commit:
Update the webpack file

itunuworks committed:
Merge pull request #33 from itunuworks/refactorBranch

Refactor branch

Commit:
feature (sources): search through sources

Implement a search feature to look through the available sources and return matches to be selected from

itunuworks committed:
Merge pull request #34 from itunuworks/feature/search-through-sources

Implement search through news sources

Commit:
chore (project): implement defense feedback chores

fix placement of dependencies
ensure correct casing in jsdoc comments
remove unused promise statement
delete unimplemented functionality from readme
include empty lines between describe blocks in tests
include project license in readme
include 'how to contribute' section in readme
remove all console.log statements

itunuworks committed:
Merge pull request #35 from itunuworks/chore/implement-defense-feedback-chores

Implement defense feedback chores

Commit:
feature (sources): allow user choose first source

This feature allows the user to choose the first news source that has its articles populated on page load

itunuworks committed:
Merge pull request #36 from itunuworks/feature/allow-user-choose-first-source

Allow user choose first source on page load

Commit:
refactor (ui/ux): redesign user interface and experience

Set up footers, color schemes and basic user interface

itunuworks committed:
Merge pull request #39 from itunuworks/feature/improve-ui-ux

Improve user interface and experience

Commit:
feature (tests): implement component tests

setup component tests and ensure coverage and 100% pass rates

Commit:
bug (login): fix login bug

fix a ui loader while the authentication is in progress and also fix the functionality of the intro text on the main page

itunuworks committed:
Merge pull request #41 from itunuworks/bug/fix-login-lag

Fix login and main page bugs

itunuworks committed:
Merge branch 'developmentBranch' into feature/implement-component-tests

Commit:
feature (tests): implement component tests

setup component tests and ensure coverage and 100% pass rates

itunuworks committed:
Merge pull request #42 from itunuworks/feature/implement-component-tests

Implement component tests

Commit:
bug (eslint): fix-eslint-errors

modified eslinting rules for better code quality and ensured compliance

itunuworks committed:
Merge pull request #43 from itunuworks/bug/fix-eslint-errors

Fix Eslint errors

Commit:
bug (project-wrapup): implement-scorecard-feedback

setup faqs and limitations section in readme
fix datatype mistakes in jsdoc
modify parameter naming
remove console.log statements from codebase
move config details to a .env file

itunuworks committed:
Merge pull request #44 from itunuworks/bug/implement-scorecard-feedback

Implement scorecard feedback

Commit:
feature (description): add a description file

THis file contains a description of the project

[finishes #147002423]

Commit:
feature (description): delete description file

This file is no more needed

[delivers #147002543]
